cmd/soroban-rpc/internal/ingest/service.go

```diff
@@ -208,9 +209,14 @@ func (s *Service) fillEntriesFromCheckpoint(ctx context.Context, archive history
 	if err != nil {
 		return err
 	}
+	transactionCommitted := false
 	defer func() {
-		if err := tx.Rollback(); err != nil {
-			s.logger.WithError(err).Warn("could not rollback fillEntriesFromCheckpoint write transactions")
+		if transactionCommitted == false {
```

Comment:
golangci - prefer `if !transactionCommitted {`

Reply:
admittedly, it's not my preference, but I'll align with golangci.